codalab/lib/bundle_util.py

```diff
@@ -354,3 +354,58 @@ def check_bundle_freezable(bundle):
             'Cannot freeze bundle %s(%s), bundle is not in a final state.'
             % (bundle.uuid, bundle.metadata.to_dict()["name"])
         )
+
+
+def get_bundle_state_details(bundle):
+    """
+    This helper returns information about what the given bundle's state means.
+    State definitions varry slightly based on the bundle's type.
+
+    If the bundle is currently running, its run status will be returned.
+    If the bundle is currently staged, its staged status will be returned.
+    """
+    if 'state' not in bundle or 'bundle_type' not in bundle:
+        return ''
+
+    type = bundle['bundle_type']
+    state = bundle['state']
+    state_details_by_type = {
```

Comment:
Maybe add a comment referencing `bundle_state.py` so that we can keep these in sync.

Reply:
Also mention "Bundle-Lifecycle.md"